metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/extractor/PublicationExtractor.java

```diff
@@ -21,35 +21,51 @@
 import uk.ac.ebi.ampt2d.metadata.persistence.entities.Publication;
 import uk.ac.ebi.ampt2d.metadata.persistence.repositories.PublicationRepository;
 import uk.ac.ebi.ena.sra.xml.LinkType;
+import uk.ac.ebi.ena.sra.xml.ProjectType;
 import uk.ac.ebi.ena.sra.xml.StudyType;
 import uk.ac.ebi.ena.sra.xml.XRefType;
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
-public class PublicationExtractorFromStudy {
+public class PublicationExtractor {
 
     private static final String PUBMED = "pubmed";
 
     private PublicationRepository publicationRepository;
 
-    public PublicationExtractorFromStudy(PublicationRepository publicationRepository) {
+    public PublicationExtractor(PublicationRepository publicationRepository) {
         this.publicationRepository = publicationRepository;
     }
 
-    public List<Publication> getPublications(StudyType.STUDYLINKS studylinks) {
-        List<Publication> publications = new ArrayList<>();
-        if (studylinks == null) {
-            return publications;
+    public List<Publication> getPublicationsFromProject(ProjectType.PROJECTLINKS projectLinks) {
+        if (projectLinks == null) {
+            return new ArrayList<>();
+        } else {
+            return getPublicationsFromLinks(
+                    Stream.of(projectLinks.getPROJECTLINKArray())
+                          .map(link -> link.getXREFLINK()));
         }
-        LinkType[] studyLinksArray = studylinks.getSTUDYLINKArray();
-        for (int i = 0; i < studyLinksArray.length; i++) {
-            XRefType xRefType = studyLinksArray[i].getXREFLINK();
-            if (xRefType != null && xRefType.getDB().equalsIgnoreCase(PUBMED)) {
-                publications.add(findOrCreatePublication(xRefType.getID()));
-            }
+    }
+
+    public List<Publication> getPublicationsFromStudy(StudyType.STUDYLINKS studylinks) {
+        if (studylinks == null) {
+            return new ArrayList<>();
+        } else {
+            return getPublicationsFromLinks(
+                    Stream.of(studylinks.getSTUDYLINKArray())
+                          .map(link -> link.getXREFLINK()));
```

Comment:
This is another case where supporting importing child elements from both study and project seemingly leads to code duplication. Maybe we should only support one way, presumably the old one — extracting everything from study?

Reply:
EVA works with projects in several places. it's going to be a constant pain having to work with studies only for calling this code.

Reply:
Sorry, I probably didn't explain myself properly. What I proposed to do was to start import from project in EVA case; then retrieve study based on that; and then do _everything_ else starting _from the study_ which we retrieved. If we do it this way, we won't have a need for `getPublicationsFromProject` and `getPublicationsFromStudy` because the publications (and all other child objects) will be retrieved from the study. It will only affect import, and we'll be able to work with both study and project accessions in the webservice

Reply:
So to clarify even further, I propose that Study becomes a central object for import for both use cases to simplify code. We will have:
* EGA case: start from Analysis → get Study → **do everything else from Study**
* EVA case: start from Project → get Study → **do everything else from Study**

Reply:
I've added this to EVA-2012